packages/reporter/src/runnables/runnables.tsx

```diff
@@ -176,11 +176,31 @@ class Runnables extends Component<RunnablesProps> {
   componentDidMount () {
     const { scroller, appState } = this.props
 
-    scroller.setContainer(this.refs.container as Element, action('user:scroll:detected', () => {
-      if (appState && appState.isRunning) {
-        appState.temporarilySetAutoScrolling(false)
-      }
-    }))
+    let maybeHandleScroll: UserScrollCallback | undefined = undefined
+
+    const containerEl = this.refs.container as HTMLElement
+
+    if (window.__CYPRESS_MODE__ === 'open') {
+      // in open mode, listen for scroll events so that users can pause the command log auto-scroll
+      // by manually scrolling the command log
+      containerEl.setAttribute('data-cy-scroll-listen', 'true')
+      maybeHandleScroll = action('user:scroll:detected', () => {
```

Comment:
Took me a while to figure out these data attrs are purely for testing. I don't think this is ideal - I think they are okay to make selecting elements less brittle, like `findByTestId(...).click()`, as opposed to `findById("#blah")`.

What I think we really want to do here is spy on the function and assert it was/was not called with the correct arguments (next best thing to actually adding a regression test, but as you pointed out this is really hard to reproduce in a reliable way).

While writing a comment on how we might test this by spying on the method, I ended up just giving it at try, as well as killing some technical debt along the way, what do you think?

https://github.com/cypress-io/cypress/pull/23774

Reply:
oh nice, this looks really good, merged!